docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+from typing import Callable, List
+
+import retrying
+
+from retrying import retry, RetryError
+from teamcity.messages import TeamcityServiceMessages
+
+import test_util.aws
+import test_util.cluster
+
+from pkgpanda.util import logger, write_string
+```
+
+* [Packages](https://google.github.io/styleguide/pyguide.html?showone=Packages#Packages)
+    * Import each module using the full pathname location of the module.
+    * No Relative Imports.
+
+* Maximum line length is 120 characters.
+    * Following are the exceptions to this rule. These types can be of arbitrary length.
+        * Long import statements.
+        * URLs in comments.
+        * Any sequence of chars which would better be in a single line.
+
+* Indentation
+
+    * Use 4space indentation.
+    * For long list of arguments, use double-indented hanging-indentation.
+ For e.g.
+    * The hanging indentation should be followed by a single blank line.
+
+ ```
+ class DcosApiSession(ARNodeApiClientMixin, ApiClientSession):
+
+     def __init__(
+             self,
+             dcos_url: str,
+             masters: list,
+             public_masters: list,
+             slaves: list,
+             public_slaves: list,
+             default_os_user: str,
+             auth_user: Optional[DcosUser]):
+
+         # ...
+         super().__init__(Url.from_string(dcos_url))
+         self.masters = sorted(masters)
+ ```
+
+
+* Adding Types. Since our code-base is Python3.5+ compatible, we have the choice of introducing types, validating with mypy.
+
+        class VpcClusterUpgradeTestDcosApiSessionFactory:
+
+            def apply(
+                    self,
+                    dcos_url: str,
+                    masters: List[str],
+                    public_masters: List[str],
+                    slaves: List[str],
+                    public_slaves: List[str],
+                    default_os_user: str) -> DcosApiSession:
+
+    * Reviewers will request the author to write types for the method parameters.
+    * We will start running mypy type checker as part of build process.
+
+* Shebang Rule.
+    * We will consistently use the shebang line in our scripts.
+        * `#!/usr/bin/env python3`
+        OR
+        * `#!/usr/bin/env python2` in our scripts.
+
+* Use the `format` method
+    * Prefer `'{named}'.format` will be over `{}.format`.
+        * Rationale: If the number of arguments are long. `{named}.format` helps in readability.
+    * Use your best judgement to decide between + and % (or format) when applicable.
+
+* TODO Comment Format
+    * Follow the rules on [when to write TODO comments](https://google.github.io/styleguide/pyguide.html?showone=TODO_Comments#TODO_Comments)
+    * Format required will be `TODO (active-developer) [JIRA] - Brief one line description`
+    * A plugin can be used to enforce the TODO format requirement.
+
+* [Comments and Docstrings](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments).
```

Comment:
For the former, maybe you missed "not" in my sentence, doesn't matter now 😄 

The latter sounds good to me!